polar-core/src/error.rs

```diff
@@ -411,7 +398,17 @@ pub enum ValidationError {
         // already-declared resource block would be relevant info for the error emitted on
         // redeclaration.
     },
-    // TODO(lm|gj): add SingletonVariable.
+    SingletonVariable {
+        /// Term<Symbol> where the error arose, tracked for lexical context.
+        term: Term,
+        /// Variable name.
+        name: String,
+    },
+    UnregisteredConstant {
+        /// Term<Symbol> where the error arose, tracked for lexical context.
+        term: Term,
+        msg: String,
```

Comment:
I think it's probably best going forward to include the information necessary to generate the message in the Error struct, but not the message itself. This way, if the error is generated at multiple sites the message will be consistent.

I realize we probably have not followed that convention so far, but maybe we can add a note to this file about it.

Reply:
I generally agree (e.g., see https://github.com/osohq/oso/commit/204d04917dfa8d8cb128567d8f348b8d92a9d8b0#diff-20681b7129ace19298fd33795d95bbf7c5b118c4be2bf33844520b21d8798d45R385-R398). I don't want to update errors per that convention as part of this PR, but I would like to standardize our errors in a follow-up.

Reply:
This is a new error though right? Can we store the constant name on it instead of the `msg`?

Reply:
We store the constant name in the `term: Term` field on line 409. The `msg` field is used to provide some unique context about the situations in which this error can arise:

https://github.com/osohq/oso/blob/17e7f43315fe19d9368bff810740f930d250c94e/polar-core/src/kb.rs#L218

https://github.com/osohq/oso/blob/17e7f43315fe19d9368bff810740f930d250c94e/polar-core/src/resource_block.rs#L650

https://github.com/osohq/oso/blob/17e7f43315fe19d9368bff810740f930d250c94e/polar-core/src/resource_block.rs#L671

https://github.com/osohq/oso/blob/17e7f43315fe19d9368bff810740f930d250c94e/polar-core/src/resource_block.rs#L687

Reply:
Actually these are a bit different... thinking about how to group them into different error types that don't require `msg` differentiation

Reply:
Oh - I didn't realize that. 🤦 Not sure it's necessary to split up then.

Reply:
Doing this cleanup as a follow-up.